subsys/sd/sd_ops.c

```diff
@@ -787,9 +794,10 @@ int card_ioctl(struct sd_card *card, uint8_t cmd, void *buf)
 		 * Note that SD stack does not support enabling caching, so
 		 * cache flush is not required here
 		 */
-		return sdmmc_wait_ready(card);
+		ret = sdmmc_wait_ready(card);
 	default:
-		return -ENOTSUP;
+		ret = -ENOTSUP;
 	}
+	k_mutex_unlock(&card->lock);
 	return 0;
```

Comment:
you changed from immediate return above to setting ret, but don't return ret, is this supposed to return ret

Reply:
Yes, will fix this